src/axom/primal/geometry/Tetrahedron.hpp

```diff
@@ -202,59 +202,45 @@ class Tetrahedron
-    using axom::numerics::dot_product;
-    using axom::utilities::abs;
-    using NumericArrayType = primal::NumericArray<T, NDIMS>;
+    constexpr double EPS = 1.0e-50;
 
     const PointType& p0 = m_points[0];
     const PointType& p1 = m_points[1];
     const PointType& p2 = m_points[2];
     const PointType& p3 = m_points[3];
 
-    // clang-format off
-    const Point<T, 4> sq { dot_product(p0.data(), p0.data(), NDIMS),
-                           dot_product(p1.data(), p1.data(), NDIMS),
-                           dot_product(p2.data(), p2.data(), NDIMS),
-                           dot_product(p3.data(), p3.data(), NDIMS)};
-
-    const double  a =  determinant(p0[0], p0[1], p0[2], 1.,
-                                   p1[0], p1[1], p1[2], 1.,
-                                   p2[0], p2[1], p2[2], 1.,
-                                   p3[0], p3[1], p3[2], 1.);
-
-    const double dx =  determinant(sq[0], p0[1], p0[2], 1.,
-                                   sq[1], p1[1], p1[2], 1.,
-                                   sq[2], p2[1], p2[2], 1.,
-                                   sq[3], p3[1], p3[2], 1.);
-
-    const double dy = -determinant(sq[0], p0[0], p0[2], 1.,
-                                   sq[1], p1[0], p1[2], 1.,
-                                   sq[2], p2[0], p2[2], 1.,
-                                   sq[3], p3[0], p3[2], 1.);
-
-    const double dz =  determinant(sq[0], p0[0], p0[1], 1.,
-                                   sq[1], p1[0], p1[1], 1.,
-                                   sq[2], p2[0], p2[1], 1.,
-                                   sq[3], p3[0], p3[1], 1.);
-
-    const double  c =  determinant(sq[0], p0[0], p0[1], p0[2],
-                                   sq[1], p1[0], p1[1], p1[2],
-                                   sq[2], p2[0], p2[1], p2[2],
-                                   sq[3], p3[0], p3[1], p3[2]);
-    // clang-format on
-
-    const auto center = NumericArrayType {dx, dy, dz} / (2 * a);
-    const T radius = sqrt(dx * dx + dy * dy + dz * dz - 4 * a * c) / (2 * abs(a));
-    return SphereType(center.data(), radius);
+    // It's useful to separate the x-, y- and z- components of
+    // the difference vectors for p1, p2, and p3 from p0
+    const VectorType vx {p1[0] - p0[0], p2[0] - p0[0], p3[0] - p0[0]};
+    const VectorType vy {p1[1] - p0[1], p2[1] - p0[1], p3[1] - p0[1]};
+    const VectorType vz {p1[2] - p0[2], p2[2] - p0[2], p3[2] - p0[2]};
+
+    // We also need their squared norms
+    const VectorType sq {vx[0] * vx[0] + vy[0] * vy[0] + vz[0] * vz[0],
+                         vx[1] * vx[1] + vy[1] * vy[1] + vz[1] * vz[1],
+                         vx[2] * vx[2] + vy[2] * vy[2] + vz[2] * vz[2]};
+
+    // Compute one over denominator using a small value to avoid division by zero
+    // Note: a has opposite sign of signedVolume()
+    const double a = VectorType::scalar_triple_product(vx, vy, vz);
+    const double ood = 1. / (2 * a - EPS);
```

Comment:
Any reason why we subtract from a very small number?

Reply:
**Re: "subtract" (vs. "add")**
I addressed this elsewhere and will rework/clean this up

**Re: the "small number" part:**
For the Delaunay computation, I was seeing cases where the volumes were really small, e.g. ~1e-20 or smaller, so this small EPS helps us avoid division by zero.